hasura/migrations/default/1742740383441_filter_open_seats/up.sql

```diff
@@ -0,0 +1,121 @@
+  -- ie) the query "ECE 105" should match both "ECE" and "105" because
+  -- the frontend will translate the raw query to "ECE:* & 105:*"
+  to_tsvector('simple', ARRAY_TO_STRING(REGEXP_MATCHES(course.code,
+    '^[a-z|A-Z]+([0-9]+[a-z|A-Z]*)'), ''))    AS document
+FROM course
+  LEFT JOIN course_prerequisite ON course_prerequisite.course_id = course.id
+  LEFT JOIN course_section ON course_section.course_id = course.id
+  LEFT JOIN materialized.prof_teaches_course ON materialized.prof_teaches_course.course_id = course.id
+  LEFT JOIN materialized.course_rating ON materialized.course_rating.course_id = course.id
+GROUP BY course.id, ratings, liked, easy, useful;
+
+CREATE VIEW course_search_index AS
+SELECT * FROM materialized.course_search_index;
+
+CREATE FUNCTION refresh_section_meeting_views()
+RETURNS TRIGGER AS $$
+  BEGIN
+    EXECUTE 'REFRESH MATERIALIZED VIEW materialized.course_rating;';
+    EXECUTE 'REFRESH MATERIALIZED VIEW materialized.prof_rating;';
+    EXECUTE 'REFRESH MATERIALIZED VIEW materialized.prof_teaches_course;';
+    EXECUTE 'REFRESH MATERIALIZED VIEW materialized.course_search_index;';
+    EXECUTE 'REFRESH MATERIALIZED VIEW materialized.prof_search_index;';
+    RETURN NULL;
+  END;
+$$ LANGUAGE plpgsql;
+
+CREATE FUNCTION search_courses(query TEXT, code_only BOOLEAN)
+RETURNS SETOF course_search_index AS $$
+  BEGIN
+    IF code_only THEN
+      RETURN QUERY
+      SELECT * FROM course_search_index
+        WHERE course_letters ILIKE query
+      ORDER BY ratings DESC;
+    ELSE
+      RETURN QUERY
+      SELECT * FROM course_search_index
+        WHERE document @@ to_tsquery('simple', query)
+      UNION
+      SELECT DISTINCT * FROM (SELECT unnest(course_ids) AS course_id
+        FROM prof_search_index
+        WHERE document @@ to_tsquery('simple', query)) prof_courses
+        LEFT JOIN course_search_index USING (course_id)
+      ORDER BY ratings DESC;
+    END IF;
+  END
+$$ LANGUAGE plpgsql STABLE;
+
+CREATE FUNCTION search_profs(query TEXT, code_only BOOLEAN)
+RETURNS SETOF prof_search_index AS $$
+  BEGIN
+    IF code_only THEN
+      RETURN QUERY
+      SELECT DISTINCT * FROM (SELECT unnest(prof_ids) AS prof_id
+        FROM course_search_index
+        WHERE course_letters ILIKE query) course_profs
+        LEFT JOIN prof_search_index USING (prof_id)
+      ORDER BY ratings DESC;
+    ELSE
+      RETURN QUERY
+      SELECT * FROM prof_search_index
+        WHERE document @@ to_tsquery('simple', query)
+      UNION
+      SELECT DISTINCT * FROM (SELECT unnest(prof_ids) AS prof_id
+        FROM course_search_index
+        WHERE document @@ to_tsquery('simple', query)) course_profs
+        LEFT JOIN prof_search_index USING (prof_id)
+      ORDER BY ratings DESC;
+    END IF;
+  END;
+$$ LANGUAGE plpgsql STABLE;
+
+CREATE TRIGGER refresh_section_meeting
+AFTER INSERT OR UPDATE OR DELETE ON section_meeting
+FOR EACH STATEMENT
+EXECUTE PROCEDURE refresh_section_meeting_views();
```

Comment:
This refreshes the view whenever `section_meeting` is updated, I think we probably also want to refresh whenever `course_section` is updated so that we update the `terms_with_seats` field properly

Reply:
Or we can swap from materialized view to normal view, probably easier